taf/repository_tool.py

```diff
@@ -146,27 +146,35 @@ def yubikey_signature_provider(name, key_id, key, data):  # pylint: disable=W061
 
     def _check_key_and_get_pin(expected_key_id):
         try:
-            inserted_key = yk.get_piv_public_key_tuf()
-            if expected_key_id != inserted_key["keyid"]:
-                return None
-            serial_num = yk.get_serial_num(inserted_key)
-            pin = yk.get_key_pin(serial_num)
+            yubikey_count = yk.check_yubikey_count()
+            if yubikey_count < 2:
+                inserted_key = yk.get_piv_public_key_tuf()
+                if expected_key_id != inserted_key["keyid"]:
+                    return None, None
+                serial = yk.get_serial_num(inserted_key)
+            else:
+                print(f"Final confirmation for {name} key")
+                serial = yk.verify_yubikey_serial()
```

Comment:
I am currently unable to test this since I am not able to insert two yubikeys at the same time, but based on the code, it looks like you have to manually enter each of the yubikey's serial number. This is probably less user-friendly compared to having to remove and insert the devices. A bigger issue, however, is that we want to be able to sign metadata using YubiKeys plugged into our server. So, this should iterate over all inserted YubiKeys and simply try to sign using all of them, instead of asking for any input. So, let's say that you have timestamp and snapshot keys inserted and you need to sign timestamp metadata. We need to automatically detect the key that can sign the timestamp metadata. At the moment, the user will have to enter pins too, but we don't need to address that in this PR.